src/pages/api/todos.js

```diff
@@ -19,7 +19,21 @@ const todos = [
 export default function handler(req, res) {
     switch (req.method) {
         case "GET":
-            res.status(200).json(todos);
+            res.status(200).json({
+                code: 200,
+                message: "todos fetched",
+                error: null,
+                data: todos,
+            });
+
+        case "DELETE":
+            todos = todos.filter((todo) => todo.id !== req.body.id);
+            res.status(200).json({
+                code: 200,
+                message: `todo with id of ${req.body.id} has been deleted!`,
+                error: null,
+                data: null,
```

Comment:
When I tried at my local, all of the lists suddenly disappear after I click the checkbox button. Is it because the data is null?

Reply:
I already solve the problem, i think it happen because i directly filtering the todos state and not the current todos state on setTodos
from : `setTodos(todos.filter((todo) => todo.id !== todoId))`
to : `setTodos((currentState) => currentState.filter((todo) => todo.id !== todoId))`